pkg/spire/spire.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spire
+
+import (
+	"context"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// KeySVID key used by TaskRun SVID
+	KeySVID = "SVID"
+	// KeySignatureSuffix is the suffix of the keys that contain signatures
+	KeySignatureSuffix = ".sig"
+	// KeyResultManifest key used to get the result manifest from the results
+	KeyResultManifest = "RESULT_MANIFEST"
+	// WorkloadAPI is the name of the SPIFFE/SPIRE CSI Driver volume
+	WorkloadAPI = "spiffe-workload-api"
+	// VolumeMountPath is the volume mount in the the pods to access the SPIFFE/SPIRE agent workload API
+	VolumeMountPath = "/spiffe-workload-api"
+)
+
+// ControllerAPIClient interface maps to the spire controller API to interact with spire
+type ControllerAPIClient interface {
+	Close() error
+	CreateEntries(ctx context.Context, tr *v1beta1.TaskRun, pod *corev1.Pod, ttl int) error
```

Comment:
This TTL is for the how long the entries that are registered into the Spire Sever are valid. Currently we are also using the default timeout setup by the config when initializing it and it will be deleted when the `tr.isDone()`. 

```
// TTL for the entry is in seconds
ttl := config.FromContextOrDefaults(ctx).Defaults.DefaultTimeoutMinutes * 60
```

Reply:
My thinking in making it a duration is that we just make the above line

```
ttl := config.FromContextOrDefaults(ctx).Defaults.DefaultTimeoutMinutes * time.Minute
```

instead of converting to seconds. wdyt?

Reply:
it does end up being translated back to `int` when talking to the SPIRE API, but am okay with either.

Reply:
Changed to `time.Duration` but like Brandon said it will have to translate back to `int` for the SPIRE API.